src/main/java/org/sonarlint/intellij/tasks/ServerDownloadProjectTask.java

```diff
@@ -42,15 +44,17 @@ public ServerDownloadProjectTask(Project project, ServerConnection server) {
 
   @Override
   protected Map<String, SonarProjectDto> compute(@NotNull ProgressIndicator indicator) throws Exception {
-    try {
-      return getService(BackendService.class)
-        .getAllProjects(server).get()
-        .getSonarProjects()
-        .stream().collect(Collectors.toMap(SonarProjectDto::getKey, p -> p));
-    } catch (Exception e) {
-      SonarLintConsole.get(myProject).error("Failed to download list of projects", e);
-      throw e;
-    }
+    return computeOnPooledThread("Download Projects", () -> {
+      try {
+        return waitForFuture(indicator, getService(BackendService.class)
```

Comment:
It would be better to put the `computeOnPooledThread` just around the `waitForFuture`, inside the try/catch

Reply:
Actually this is not that better, ideally the catch should be outside, as it is catching any exception and it returns the error message to the dialog in the Wizard (so that the user can understand). But currently, the computeOnPooledThread is catching any exception as well and just returns null. I think it would be best if we add a new computeOnPooledThreadWithoutCatching, something like that, that would use a custom waitForTask that would throw the exception and not just catch it, so it can go back to the original call.
Probably a similar issue in other xxxTask where we used computeOnPooledThread